src/Application/Rokalo.Application/Services/PasswordHashingService.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace Rokalo.Application.Services
+{
+    using System;
+    using System.Security.Cryptography;
+    using System.Text;
+
+    internal interface IPasswordHashingService
+    {
+        string HashPassword(string password, out byte[] salt);
+
+        bool VerifyPassword(string password, string hash, byte[] salt);
+    }
+
+    internal class PasswordHashingService : IPasswordHashingService
+    {
+        const int keySize = 64;
+        const int iterations = 350000;
+        HashAlgorithmName hashAlgorithm = HashAlgorithmName.SHA512;
+
+        public string HashPassword(string password, out byte[] salt)
```

Comment:
https://andrewlock.net/exploring-the-asp-net-core-identity-passwordhasher/

`private static byte[] HashPasswordV2(string password, RandomNumberGenerator rng)
{
    const KeyDerivationPrf Pbkdf2Prf = KeyDerivationPrf.HMACSHA1; // default for Rfc2898DeriveBytes
    const int Pbkdf2IterCount = 1000; // default for Rfc2898DeriveBytes
    const int Pbkdf2SubkeyLength = 256 / 8; // 256 bits
    const int SaltSize = 128 / 8; // 128 bits

    // Produce a version 2 text hash.
    byte[] salt = new byte[SaltSize];
    rng.GetBytes(salt);
    byte[] subkey = KeyDerivation.Pbkdf2(password, salt, Pbkdf2Prf, Pbkdf2IterCount, Pbkdf2SubkeyLength);

    var outputBytes = new byte[1 + SaltSize + Pbkdf2SubkeyLength];
    outputBytes[0] = 0x00; // format marker
    Buffer.BlockCopy(salt, 0, outputBytes, 1, SaltSize);
    Buffer.BlockCopy(subkey, 0, outputBytes, 1 + SaltSize, Pbkdf2SubkeyLength);
    return outputBytes;
}`

Reply:
Isn't V2 also old hash format and we should use V3?